pygments/lexers/savi.py

```diff
@@ -0,0 +1,154 @@
+"""
+    pygments.lexers.savi
+    ~~~~~~~~~~~~~~~~~~~~
+
+    Lexer for Savi.
+
+    :copyright: Copyright 2006-2021 by the Pygments team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+from pygments.lexer import RegexLexer, bygroups
+from pygments.token import *
+
+__all__ = ['SaviLexer']
+
+# The canonical version of this file can be found in the following repository,
+# where it is kept in sync with any language changes, as well as the other
+# pygments-like lexers that are maintained for use with other tools:
+# - https://github.com/savi-lang/savi/blob/main/tooling/pygments/lexers/savi.py
+#
+# If you're changing this file in the pygments repository, please ensure that
+# any changes you make are also propagated to the official Savi repository,
+# in order to avoid accidental clobbering of your changes later when an update
+# from the Savi repository flows forward into the pygments repository.
+#
+# If you're changing this file in the Savi repository, please ensure that
+# any changes you make are also reflected in the other pygments-like lexers
+# (rouge, vscode, etc) so that all of the lexers can be kept cleanly in sync.
+
+class SaviLexer(RegexLexer):
+  """
+  For `Savi <https://github.com/savi-lang/savi>`_ source code.
+  """
+
+  name = 'Savi'
+  aliases = ['savi']
+  filenames = ['*.savi']
+
+  root_rules = [
+    # Line Comment
+    (r'//.*?$', Comment.Singleline),
+
+    # Doc Comment
+    (r'::.*?$', Comment.Singleline),
+
+    # Capability Operator
+    (r'(\')(\w+)(?=[^\'])', bygroups(Operator, Name)),
+
+    # Double-Quote String
+    (r'\w?"', String.Double, "string.double"),
```

Comment:
Please use `+`  and not `?` so multiple tokens get matched together.

Reply:
This is actually a single optional `\w` character accepted here, not multiple - note that this is immediately before the string, not inside the string.

The optional `\w` here enables things like `b"example"` for a byte string, or `r"example"` for a regular expression string (similar to the one seen here in the python source code).

Reply:
Ah, my bad. I'm looking at this here in the output:
```
'"'           Literal.String.Double
's'           Literal.String.Double
'p'           Literal.String.Double
'e'           Literal.String.Double
'c'           Literal.String.Double
'"'           Literal.String.Double
```

And those `s`, `p`, `e`, `c` should become one token. I just scanned for `String.Double` and found the wrong rule :)

Reply:
Yes, I've fixed that issue in my latest commit.